collections.js

```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
     node = node.next;
   }
   return null;
+};
+
+//------------------------------------------------------------------------------
+// Simple Queue.
+
+function SimpleQueue() {
+	this.q_ = [];
 }
 
+SimpleQueue.prototype = {
+  enqueue: function(item) {
+    this.q_.push(item);
+    return this;
+  },
+
+  dequeue: function() {
+    return this.q_.shift();
+  },
+
+  empty: function() {
+    return this.q_.length === 0;
+  },
+  
+  flush: function() {
+    var result = this.q_.slice(0);
+	this.q_.length = 0;
+	return result;
```

Comment:
Why not just reassign the array? Also, it's strange to return the array.

  this.q_ = [];
  return this;